src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -226,6 +226,24 @@ struct task_metrics {
   }
 };
 
+enum class oom_type {
+  CPU_OR_GPU = 0,
+  CPU,
+  GPU,
+};
+
+struct oom_state_type {
```

Comment:
should this just be `oom_state`?

Reply:
IMO in snake case it becomes cumbersome when you want to have a local variable

camel case is fine `OomState oomState` but `oom_state oom_state` is not